packages/modules/app/src/app/flows.ts

```diff
@@ -93,6 +93,75 @@ export const handleFetchConfig =
             }),
         );
 
+/**
+ * Handles the fetch settings action by fetching the app settings from the provider,
+ * filtering out null values, and dispatching success or failure actions accordingly.
+ *
+ * @param provider The AppModuleProvider used to fetch the app settings.
+ * @returns A Flow function that takes an Observable of actions and returns an Observable of actions.
+ */
+export const handleFetchSettings =
+    (provider: AppModuleProvider): Flow<Actions, AppBundleState> =>
+    (action$) =>
+        action$.pipe(
+            // only handle fetch settings request actions
+            filter(actions.fetchSettings.match),
+            // when request is received, abort any ongoing request and start new
+            switchMap((action) => {
+                const { payload: appKey } = action;
+
+                // fetch settings from provider
+                const subject = from(provider.getAppSettings(appKey)).pipe(
+                    // filter out null values
+                    filter((x) => !!x),
+                    // allow multiple subscriptions
+                    share(),
+                );
+
+                // first load settings and then dispatch success action
+                return concat(
+                    subject.pipe(map((settings) => actions.setSettings(settings))),
+                    subject.pipe(
+                        last(),
+                        map((settings) => actions.fetchSettings.success(settings)),
+                    ),
+                ).pipe(
+                    // catch any error and dispatch failure action
+                    catchError((err) => {
+                        return of(actions.fetchSettings.failure(err));
+                    }),
+                );
+            }),
+        );
+
+/**
+ * Handles the set settings action by setting the app settings from the provider,
+ * filtering out null values, and dispatching success or failure actions accordingly.
+ *
+ * @param provider The AppModuleProvider used to fetch the app settings.
+ * @returns A Flow function that takes an Observable of actions and returns an Observable of actions.
+ */
+export const handleUpdateSettings =
```

Comment:
update settings must allow to partial patch settings, so this flow either needs to merge state settings, or the provider needs to merge settings from provider. Also we need to assure that this flow is run in sequnece to prevent override of pending setting changes